content/blog/2019/07/2019-07-03-remoting-kafka-kubernetes-phase-1.adoc

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "Remoting over Apache Kafka plugin with Kafka launcher in Kubernetes"
+tags:
+- remoting
+- kafka
+- kubernetes
+- helm
+- developer
+- gsoc
+- gsoc2019
+author: longnguyen
+---
+
+= Remoting over Apache Kafka plugin with Kafka launcher in Kubernetes
+
+== Introduction
+
+Current version of Remoting over Apache Kafka plugin requires users to manually configure the entire system which includes Zookeeper, Kafka and remoting agents. It also doesn't support dynamic agent provisioning so scalability is harder to achieve. This project aims to solve two problems:
+
+. Out-of-the-box solution to provision Apache Kafka cluster.
+
+. Dynamic agent provisioning in a Kubernetes cluster.
+
+== Current State
+
+* Kubernetes connector with credentials supported.
+
+* Apache Kafka provisioning in Kubernetes feature is fully implemented.
+
+* Helm chart is partially implemented.
+
+== Apache Kafka provisioning in Kubernetes
+
+This feature is part of 2.0 version so it is not yet released officially. You can try out the feature by building directly from master branch:
+
+[source, bash]
+----
+git clone https://github.com/jenkinsci/remoting-kafka-plugin.git
+cd remoting-kafka-plugin/plugin
+mvn hpi:run
+----
+
+On the Global Configuration page, users can input Kubernetes server information and credentials. Then they can start Apache Kafka with only one button click.
+
+image:/images/post-images/remoting-kafka/kafka-provisioning-kubernetes-ui.png[Kafka provisioning in Kubernetes UI]
+
+When users click `Start Kafka on Kubernetes` button, Jenkins will create a Kubernetes client from the information and then apply Zookeeper and Kafka YAML specification files from `resources`.
+
+image:/images/post-images/remoting-kafka/kafka-provisioning-kubernetes-architecture.png[Kafka provisioning in Kubernetes architecture]
+
+== Helm Chart
+
+Helm chart for Remoting over Apache Kafka plugin is based on link:https://github.com/helm/charts/tree/master/stable/jenkins[stable/jenkins] chart and link:https://github.com/helm/charts/tree/master/incubator/kafka[incubator/kafka] chart. As of now, the chart is still a link:https://github.com/jenkinsci/remoting-kafka-plugin/pull/62[Work in Progress] because it is still waiting for Cloud API implementation in Phase 2. However, you can check out the demo chart with a single standalone Remoting Kafka Agent:
+
+[source, bash]
+----
+git clone -b demo-helm-phase-1 https://github.com/longngn/remoting-kafka-plugin.git
+cd remoting-kafka-plugin
+K8S_NODE=<your Kubernetes node IP> ./helm/jenkins-remoting-kafka/do.sh start
+----
+
+The command `do.sh start` will install the chart (with Jenkins and Kafka), launch a Kafka computer on Jenkins master by applying JCasC, then launch a single Remoting Kafka Agent pod.
```

Comment:
would be nice to expand JCasC here (reference the plugin and automatic YAML generation?)